pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/StageConditionalTranslator.groovy

```diff
@@ -62,18 +66,24 @@ public class StageConditionalTranslator implements MethodMissingWrapper, Seriali
         }
 
         if (Utils.instanceOfWrapper(Closure.class, argVal)) {
-            argVal = createStepsBlock(argVal)
-        }
-
-        DeclarativeStageConditionalDescriptor descriptor = DeclarativeStageConditionalDescriptor.byName(s)
-        if (descriptor == null) {
-            throw new NoSuchMethodError(Messages.ModelValidator_ModelASTWhen_unknown(s, DeclarativeStageConditionalDescriptor.allNames().join(", ")))
-        }
+            if (Utils.nestedWhenCondition(s)) {
```

Comment:
Wait, check that, found a reason - `DescribableStageConditionalDescriptor` isn't serializable, and as far as I can tell, most (if not all) descriptors aren't serializable so it'd be a bit weird to make this one so. As a result, we can't actually have an instance of a `DescribableStageConditionalDescriptor` in CPS code. Lemme think about caching instead.